test/spatial/diffusion.jl

```diff
@@ -59,20 +59,42 @@ Nsims = 10000
 rel_tol = 0.01
 times = 0.0:tf/num_time_points:tf
 
+algs = [NSM(), DirectCRDirect()]
 grids = [CartesianGridRej(dims), CartesianGridIter(dims), LightGraphs.grid(dims)]
-jump_problems = JumpProblem[JumpProblem(prob, NSM(), majumps, hopping_constants=hopping_constants, spatial_system = grid, save_positions=(false,false)) for grid in grids]
-push!(jump_problems, JumpProblem(prob, DirectCRDirect(), majumps, hopping_constants=hopping_constants, spatial_system = grids[1], save_positions=(false,false)))
-# setup flattenned jump prob
-graph = LightGraphs.grid(dims)
-push!(jump_problems, JumpProblem(prob, NRM(), majumps, hopping_constants=hopping_constants, spatial_system = graph, save_positions=(false,false)))
+jump_problems = JumpProblem[JumpProblem(prob, rand(algs), majumps, hopping_constants=hopping_constants, spatial_system = grid, save_positions=(false,false)) for grid in grids]; sizehint!(jump_problems, 10)
```

Comment:
`rand(algs)` so that tests don't take twice longer than they do. If that's bad, I can change it to use both.
What is the benefit of `StableRngs` with a fixed seed? To avoid the very small probability that tests don't get passed because of a 1 in a million fluke?

Reply:
Yes, let's test all the algorithms for now. If it starts taking too long we can change it to just use a specific fixed one of them for most tests (I wouldn't want to use a random algorithm in any case).

The benefit of using a fixed seed and StableRNG is it makes the test deterministic, so if it suddenly breaks we know it is definitely a code issue. For your tests it probably doesn't matter, but we've definitely had issues with other tests occasionally breaking because of the rng hitting a degenerate case.